modules/library/main/src/main/java/org/geotools/feature/simple/SimpleFeatureImpl.java

```diff
@@ -374,6 +374,11 @@ public boolean isNillable() {
         return userData;
     }
 
+    @Override
+    public boolean hasUserData() {
+        return userData != null;
```

Comment:
A question: Omitting the map.isEmpty() check is intended?  Overrided default method does the check but here only a null check is done.

Reply:
Good question, my intention was just to avoid unecessary allocations on each single feature going through the renderer, but indeed, the semantic of the method calls for checking isEmpty too. Fixing it.